src/js/api/jwplayer.api.js

```diff
@@ -333,21 +337,44 @@
 
 
         _this.remove = function() {
-            if (!_playerReady) {
-                throw 'Cannot call remove() before player is ready';
+
+            // Cancel embedding even if it is in progress
+            if (this._embedder && this._embedder.destroy) {
+                this._embedder.destroy();
             }
-            _remove(this);
-        };
 
-        function _remove(player) {
             _queuedCalls = [];
 
-            if (_embedder && _embedder.destroy) {
-                _embedder.destroy();
+            // Is there more than one player using the same DIV on the page?
+            var sharedDOM = (_.size(_.where(_players, {id : _this.id})) > 1);
+
+            // If sharing the DOM element, don't reset CSS
+            if (! sharedDOM) {
+                utils.clearCss('#' + _this.id);
             }
 
-            jwplayer.api.destroyPlayer(player.id);
-        }
+            var toDestroy = document.getElementById(_this.id + (_this.renderingMode === 'flash' ? '_wrapper' : ''));
+
+            if (toDestroy) {
+                if (_this.renderingMode === 'html5') {
+                    // calls jwPlayerDestroy()
+                    _this.destroyPlayer();
+                }
+
+                // If the tag is reused by another player, do not destroy the div
+                if (! sharedDOM) {
+                    var replacement = document.createElement('div');
```

Comment:
It will always be a div, since the original element is trashed, copying over only the id. We could do this as a forward compatibility precaution, but it may be confusing.